lib/routes/charge/pos_payment_dialog.dart

```diff
@@ -82,12 +97,25 @@ class _PosPaymentDialogState extends State<PosPaymentDialog> {
         }
       });
     });
+
+    _nfcInvoiceSubscription =
+        widget._lnUrlBloc.nfcWithdrawStream.listen(_listenNfcWithdraw);
+
+    widget._lnUrlBloc.actionsSink.add(RegisterNfcSaleRequest(
```

Comment:
isn't it better to send the paymentRequest along with the sale request? It ensures we use the same invoice in the QR and the NFC and eliminate the need to create another invoice in the bloc.

Also I think we shouldn't count on the NfcWithdrawInvoiceStatus.completed status to signal payment succeed but we should wait for the invoice to be paid as we do with the QR which is another reason why to use the same bolt11 payment request.